<launch>
    <arg name="sim" default="false"/>
    <arg name="dummyball" default="false"/>
    <arg name="debug" default="false"/>
    <arg name="dyn" default="true"/>

    <!-- start vision for wolfgang (fcnn)-->
    <group unless="$(arg dummyball)">
        <node pkg="bitbots_vision" type="vision.py" name="bitbots_vision" output="screen">
            <rosparam command="load" file="$(find bitbots_vision)/config/visionparams.yaml"/>
            <param name="vision_debug_image_msg" value="$(arg debug)"/>
            <param name="vision_ball_classifier" value="fcnn"/>
            <param name="vision_dynamic_color_space" value="$(arg dyn)"/>
        </node>
        <!-- start dynamic-color-space node -->
        <node pkg="bitbots_vision" type="dynamic_color_space.py" name="bitbots_dynamic_color_space" output="screen">
            <rosparam command="load" file="$(find bitbots_vision)/config/bitbots_dynamic_color_space_params.yaml"/>
        </node>
    </group>

    <!-- start vision without balldetection-->
    <group if="$(arg dummyball)">
        <node pkg="bitbots_vision" type="vision.py" name="bitbots_vision" output="screen">
            <rosparam command="load" file="$(find bitbots_vision)/config/visionparams.yaml"/>
            <param name="vision_debug_image_msg" value="$(arg debug)"/>
            <param name="vision_ball_classifier" value="dummy"/>
            <param name="vision_dynamic_color_space" value="$(arg dyn)"/>
        </node>
        <!-- start dynamic-color-space node -->
        <node pkg="bitbots_vision" type="dynamic_color_space.py" name="bitbots_dynamic_color_space" output="screen">
            <rosparam command="load" file="$(find bitbots_vision)/config/bitbots_dynamic_color_space_params.yaml"/>
        </node>
    </group>

    <!-- start the camera only when necessary -->
    <group unless="$(arg sim)">
        <include file="$(find wolves_image_provider)/launch/image_provider.launch" />
    </group>

    <!-- remap stuff to fit the robot -->
    <group if="$(arg sim)">
        <param name="use_sim_time" value="true"/>
        <remap to ="/camera/image_raw" from="/image_raw"/>
    </group>      
</launch>
